rsfbclient-diesel/src/fb/connection.rs

```diff
@@ -0,0 +1,194 @@
+//! The Firebird connection
+
+use super::backend::Fb;
+use super::query_builder::FbQueryBuilder;
+use super::transaction::FbTransactionManager;
+use super::value::FbRow;
+use diesel::connection::*;
+use diesel::deserialize::*;
+use diesel::expression::QueryMetadata;
+use diesel::query_builder::bind_collector::RawBytesBindCollector;
+use diesel::query_builder::*;
+use diesel::result::Error::DatabaseError;
+use diesel::result::Error::DeserializationError;
+use diesel::result::*;
+use rsfbclient::SimpleConnection as FbRawConnection;
+use rsfbclient::{Execute, Queryable, SqlType};
+use std::cell::RefCell;
+
+pub struct FbConnection<'c> {
+    pub raw: RefCell<FbRawConnection>,
+    tr_manager: FbTransactionManager<'c>,
+}
```

Comment:
The main problem seems to be here, as the `tr_manager` needs to hold a reference to the `raw`, making it a self referential struct, so this `<'c>` lifetime does not make sense, making that unsafe block in the `begin_transaction` create an potentially invalid lifetime (because it creates a lifetime that matches `'c` with unsafe code)